subsys/bluetooth/host/conn.c

```diff
@@ -1062,6 +1062,24 @@ void bt_conn_cb_register(struct bt_conn_cb *cb)
 	callback_list = cb;
 }
 
+void bt_conn_cb_unregister(struct bt_conn_cb *ucb)
+{
+	struct bt_conn_cb *cb, *prev_cb;
+
+	for (cb = callback_list, prev_cb = NULL; cb; cb = cb->_next) {
```

Comment:
Perhaps we should convert to use sys_slist_t instead of keeping the custom list implementation, it hasn't been a problem for now since we didn't have to remove items but now that are adding we got even more duplicated maybe it is work changing it.

Reply:
I was just going to say the same. The struct size will stay the same and there's no need to go implementing linked list manipulation routines ourselves.

Reply:
Adding and removing should also be protected, since these APIs can be called from a preemptible thread, and callbacks can be called from RX thread.

Reply:
I agree with your comments. I will check whether I have time for doing this now - in the context of this PR - or if we should create an issue on it.